tensordict/tensordict.py

```diff
@@ -4209,12 +4234,48 @@ def __init__(
                 f"got {type(source)}"
             )
         self._source = source
-        idx = (idx,) if not isinstance(idx, (tuple, list, range)) else tuple(idx)
+        idx = (
+            (idx,)
+            if not isinstance(
+                idx,
+                (
+                    tuple,
+                    list,
+                ),
```

Comment:
Why did `range` get removed here?

Reply:
bc we dont need to convert it to a tuple and it's a bit crazy to do `idx = tuple(range(1_000_000))` no?